controllers/compaction/reconciler.go

```diff
@@ -260,7 +260,7 @@ func (r *Reconciler) delete(ctx context.Context, logger logr.Logger, etcd *druid
 func (r *Reconciler) createCompactionJob(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd) (*batchv1.Job, error) {
 	activeDeadlineSeconds := r.config.ActiveDeadlineDuration.Seconds()
 
-	_, etcdBackupImage, err := utils.GetEtcdImages(etcd, r.imageVector)
+	_, etcdBackupImage, err := utils.GetEtcdImages(etcd, r.imageVector, r.config.FeatureGates[string(features.UseEtcdWrapper)])
```

Comment:
Not directly related to this PR but since you are making related changes i will propose one small change.
In each Config we have `FeatureGates map[string]bool`. Instead you can use `FeatureGates map[featuregate.Feature]bool` This then allows you do just do this for the above line of code:
```go
utils.GetEtcdImages(etcd, r.imageVector, r.config.FeatureGates[features.UseEtcdWrapper])
```
Casting to string is quite unnecessary specially when `featuregate.Feature` is itself is of type string.